### PR TITLE
Add support for git-secret

### PR DESCRIPTION
**Pull request by ab77**
* checks for preset keys and tooling
* reveal secrets using git-secret.io workflow
* fix build args processing
* add Docker BuikdKit secrets handling (compatible with balena-secrets)

Change-type: patch

### CODE REVIEW

File: docker/build.sh [hunk 1]
```diff
@@ -64,62 +99,94 @@ function image_variant() {
...
+      "${DOCKERFILE}" \
+      --progress=plain \
+      --cache-from "$(image_variant "${DOCKER_IMAGE}" "${sha}")" \
+      --cache-from "$(image_variant "${DOCKER_IMAGE}" "build-${branch}")" \
+      --cache-from "$(image_variant "${DOCKER_IMAGE}" master)" \
       ${args} \
-      --build-arg RESINCI_REPO_COMMIT=${sha} \
+      --build-arg RESINCI_REPO_COMMIT="${sha}" \
       --build-arg CI=true \
-      --build-arg NPM_TOKEN=${NPM_TOKEN} \
```
Comment (ab77) [inline docker/build.sh:91]: @nazrhom I've added the parallel NPM_TOKEN secret as per your PR.

**Comment by ab77** [comment]
@balena-ci retest

File: docker/build.sh [hunk 2]
```diff
@@ -64,62 +99,94 @@ function image_variant() {
...
+function build_with_opts() {
+    local dockerfile=$1; shift
+    local build_stages
+    build_stages="$(grep -Ei '^FROM\s+.*\s+AS\s+.*$' "${dockerfile}" | tr '[:upper:]' '[:lower:]' | awk -F' as ' '{print $2}')"
+
+    if [[ ${build_stages} == '' ]]; then
+        # normal build
+        docker build "$@"
+    else
+        # multistage build handling
```
Comment (robertgzr) [inline docker/build.sh:117]: :tada: for buildkit and balena-secrets compat
do you have an example for such a multi-stage build?

Comment (ab77) [reply docker/build.sh:117]: @robertgzr https://github.com/balenaci/e2e/pull/1

**Comment by ab77** [comment]
@balena-ci I self-certify!

**Pull request merged**